src/Geno/C++/Components/Project.cpp

```diff
@@ -113,156 +138,112 @@ void Project::Build( ICompiler& rCompiler )
 
 //////////////////////////////////////////////////////////////////////////
 
+static void SerializeChildren( const INode*& rNode, jsonSerializer& rSerializer, std::vector< std::string >& rFilesArray )
```

Comment:
Any particular reason `rNode` here is a reference to a pointer, and not just a `const INode* pNode` ? Reference to pointer is usually redundant unless you intend on actually changing what the passed-in original variable was pointing to!